libglusterfs/src/inode.c

```diff
@@ -1310,9 +1321,11 @@ inode_rename(inode_table_t *table, inode_t *srcdir, const char *srcname,
 
     pthread_mutex_lock(&table->lock);
     {
-        __inode_link(inode, dstdir, dstname, iatt, hash);
+        linked_inode = __inode_link(inode, dstdir, dstname, iatt, hash);
         /* pick the old dentry */
-        dentry = __inode_unlink(inode, srcdir, srcname);
+        if (linked_inode) {
```

Comment:
Been sometime I tested this. But without this fix, there was a crash when a cross namespace rename was attempted. (this may be due to the review comment you gave on line 212. Will retest).  I also added this, because there are many cases of `return NULL;` in __inode_link(), in which case, I felt we shouldn't proceed with __inode_unlink() of the source. (Can this be a different bug)

Reply:
My comment was not meant to remove this check, but if this check fails, can we continue without returning an error ?

However, after second thought I think it's fine.

Reply:
Will keep it as is, but add a reference about this comment. as a comment in the code.